hotshot-query-service/src/data_source/storage/sql/transaction.rs

```diff
@@ -496,8 +496,9 @@ where
         // Similarly, we can initialize the payload table with a null payload, which can help us
         // distinguish between blocks that haven't been produced yet and blocks we haven't received
         // yet when answering queries.
-        self.upsert("payload", ["height"], ["height"], [(height as i64,)])
-            .await?;
+        let query = query("INSERT INTO payload (height) VALUES ($1) ON CONFLICT DO NOTHING")
```

Comment:
I feel like this needs a comment. Ideally also a regression test that fails on `main` and passes with this change.

Reply:
the issue comes up only in deployment for PoS, so unfortunately, its not easy to add a test